### PR TITLE
Added reservation recommendations and tags

### PR DESCRIPTION
**Pull request by asarkar84**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json [hunk 1]
```diff
@@ -533,6 +566,55 @@
...
+        },
+        "x-ms-examples": {
+          "ReservationRecommendations": {
+            "$ref": "./examples/ReservationRecommendationsList.json"
+          },
+          "ReservationRecommendationsFilterByScopeLookBackPeriod": {
+            "$ref": "./examples/ReservationRecommendationsFilterByScopeLookBackPeriod.json"
+          }
+        },
+        "parameters": [
```
Comment (jhendrixMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: The ```subscriptionId``` parameter is missing which is the cause of the travis failure, see the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/354117627).

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json [hunk 2]
```diff
@@ -533,6 +566,55 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Consumption/reservationRecommendations": {
+      "get": {
+        "tags": [
+          "ReservationRecommendations"
+        ],
+        "operationId": "reservationRecommendations_List",
```
Comment (jhendrixMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Should be "ReservationRecommendations_List" (upper-case 'r').

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/ReservationRecommendationsList.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            "location": "northeurope",
+            "properties": {
+              "lookBackPeriod": "Last7Days",
+              "meterId": "00000000-0000-0000-0000-000000000000",
+              "term": "P1Y",
+              "costWithNoReservedInstances": 0.0,
+              "recommendedQuantity": 1,
+              "totalCostWithReservedInstances": 0.0,
+              "netSavings": 4.634521202630137,
+              "firstUsageDate": "2018-03-06",
```
Comment (jhendrixMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/ReservationRecommendationsList.json]: The schema defines this as date/time however you've only provided a date; this is causing model validation failures, see the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/354117637).  This is applicable to other samples as well.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Also note that the addition of the '$apply' parameter will introduce a breaking change for some SDKs.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json (hunk 1 above)
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Sure! Just missed it. I have added it.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json (hunk 2 above)
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Sure! Corrected.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/ReservationRecommendationsList.json (hunk 3 above)
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/ReservationRecommendationsList.json]: Thanks! for sharing this. I have changed the format to Date.

**Comment by asarkar84** [comment]
Review comments have been incorporated. Please take a look now.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json [hunk 4]
```diff
@@ -1550,6 +1635,99 @@
...
+        "netSavings": {
+          "description": "Total estimated savings with reserved instances.",
+          "type": "number",
+          "format": "decimal",
+          "readOnly": true
+        },        
+        "firstUsageDate": {
+          "description": "The usage date for looking back.",
+          "type": "string",
+          "format": "date-time",
```
Comment (jhendrixMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: There are still model validation failure regarding this date-time, see the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/354454455).  Is this supposed to be just date like the others you changed?

Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Sorry! Typo error. There was a space in the datetime string example. Just corrected it and waiting for build to complete.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/ReservationRecommendationsList.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            "location": "northeurope",
+            "properties": {
+              "lookBackPeriod": "Last7Days",
+              "meterId": "00000000-0000-0000-0000-000000000000",
+              "term": "P1Y",
+              "costWithNoReservedInstances": 0.0,
+              "recommendedQuantity": 1,
+              "totalCostWithReservedInstances": 0.0,
+              "netSavings": 4.634521202630137,
+              "firstUsageDate": "2018-03-06T00:00:00Z ",
```
Comment (jhendrixMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/ReservationRecommendationsList.json]: Another extra whitespace, looks like there's on line 42 as well.  Please scrub the remaining example files for this error.

Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/ReservationRecommendationsList.json]: Done! I have verified the other files as well.

**Comment by jhendrixMSFT** [comment]
@AutorestCI rebuild azure-sdk-for-go

**Comment by asarkar84** [comment]
Please merge the PR if everything looks good. Thank you!

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**